mmv1/templates/terraform/decoders/vertex_ai_models.go.tmpl

```diff
@@ -0,0 +1,10 @@
+// Handle the name field not being set in the API response after Create (is set after Read)
+if _, ok := res["name"]; !ok {
+	if v, ok := res["model"]; ok {
+		res["name"] = v.(string)
+	}
+}
+
+res["modelName"] = res["displayName"]
```

Comment:
Why do we need to rename displayName to modelName? Why can't mmv1/products/vertexai/Model.yaml contain a field called modelName?

Reply:
the `modelName` field is non-existent in the API.

At first I was using displayName but found that it would be best to rename it to be `modelName` in order for it to be more clear for users that are viewing the state.

It also makes it easier to understand that `source_model` is for COPYING and `modelName` is for UPLOADING since `modelName` is required when uploading a model. It is set as required in the API.
`Required. The display name of the Model. The name can be up to 128 characters long and can consist of any UTF-8 characters.`

https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.models

In the case of COPYING, we grab the existing modelName from the model that's being copied and set the `modelId` if applied in tfconfig. If not applied it will be autogenerated.

Reply:
@BBBmau sorry, I mistyped my first message and I've updated it. My question was why we would need to rename a field from the API.

Could we avoid renaming the field in this PR, as this PR is only concerned with create-via-copy? If we rename the field in this PR we give the false impression that the rename is required for a technical reason/limitation. If it's change chosen for just a user experience decision I think that should be discussed on the PR for create via upload

Reply:
ah in this case I'll revert it to use displayName since you're correct that it's not a requirement for this PR to rename the field from the API.

But it is worth knowing noting that the use of modelName is meant for :UPLOAD and will be renamed once :UPLOAD PR is merged into here. I can add this in the description of this PR.